File: app/src/main/java/com/i2medier/financialpro/adapter/RefinanceAdapter.java
package com.i2medier.financialpro.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.i2medier.financialpro.R;
import com.i2medier.financialpro.model.MonthModel;
import com.i2medier.financialpro.util.Utils;

import java.util.ArrayList;


public class RefinanceAdapter extends RecyclerView.Adapter<RefinanceAdapter.ViewHolder> {
    Context context;
    ArrayList<MonthModel> monthModels;

    public RefinanceAdapter(Context context, ArrayList<MonthModel> arrayList) {
        this.context = context;
        this.monthModels = arrayList;
    }

    public void setList(ArrayList<MonthModel> arrayList) {
        this.monthModels = arrayList;
        notifyDataSetChanged();
    }

    public ArrayList<MonthModel> getMonthModels() {
        return this.monthModels;
    }

    @Override
    @NonNull
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        return new ViewHolder(LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_month, viewGroup, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int i) {
        viewHolder.txtMonth.setText(String.valueOf(this.monthModels.get(i).getYear()));
        viewHolder.txtPrincipal.setText(Utils.decimalFormat.format(this.monthModels.get(i).getPrincipalAmount()));
        viewHolder.txtInterest.setText(Utils.decimalFormat.format(this.monthModels.get(i).getInterest()));
        viewHolder.txtPaid.setText(Utils.decimalFormat.format(this.monthModels.get(i).getTotalPaid()));
        viewHolder.txtBalance.setText(Utils.decimalFormat.format(this.monthModels.get(i).getBalance()));
        if (i % 2 == 0) {
            viewHolder.llMain.setBackgroundColor(this.context.getResources().getColor(R.color.colorWhite));
        } else {
            viewHolder.llMain.setBackgroundColor(this.context.getResources().getColor(R.color.colorLight));
        }
    }

    @Override
    public int getItemCount() {
        return this.monthModels.size();
    }


    public class ViewHolder extends RecyclerView.ViewHolder {
        LinearLayout llMain;
        TextView txtBalance;
        TextView txtInterest;
        TextView txtMonth;
        TextView txtPaid;
        TextView txtPrincipal;

        public ViewHolder(@NonNull View view) {
            super(view);
            this.llMain = (LinearLayout) view.findViewById(R.id.llMain);
            this.txtMonth = (TextView) view.findViewById(R.id.txtMonth);
            this.txtPrincipal = (TextView) view.findViewById(R.id.txtPrincipal);
            this.txtInterest = (TextView) view.findViewById(R.id.txtInterest);
            this.txtPaid = (TextView) view.findViewById(R.id.txtPaid);
            this.txtBalance = (TextView) view.findViewById(R.id.txtBalance);
        }
    }
}

File: app/src/main/java/com/i2medier/financialpro/adapter/RefinanceAdapter.kt
package com.i2medier.financialpro.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.i2medier.financialpro.R
import com.i2medier.financialpro.model.MonthModel
import com.i2medier.financialpro.util.Utils

class RefinanceAdapter(
    private val context: Context,
    var monthModels: ArrayList<MonthModel>
) : RecyclerView.Adapter<RefinanceAdapter.ViewHolder>() {

    fun setList(arrayList: ArrayList<MonthModel>) {
        this.monthModels = arrayList
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_month, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val model = monthModels[position]
        holder.txtMonth.text = model.year.toString()
        holder.txtPrincipal.text = Utils.decimalFormat.format(model.PrincipalAmount)
        holder.txtInterest.text = Utils.decimalFormat.format(model.Interest)
        holder.txtPaid.text = Utils.decimalFormat.format(model.TotalPaid)
        holder.txtBalance.text = Utils.decimalFormat.format(model.Balance)
        holder.llMain.setBackgroundColor(
            if (position % 2 == 0) ContextCompat.getColor(context, R.color.colorWhite)
            else ContextCompat.getColor(context, R.color.colorLight)
        )
    }

    override fun getItemCount(): Int = monthModels.size

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val llMain: LinearLayout = itemView.findViewById(R.id.llMain)
        val txtMonth: TextView = itemView.findViewById(R.id.txtMonth)
        val txtPrincipal: TextView = itemView.findViewById(R.id.txtPrincipal)
        val txtInterest: TextView = itemView.findViewById(R.id.txtInterest)
        val txtPaid: TextView = itemView.findViewById(R.id.txtPaid)
        val txtBalance: TextView = itemView.findViewById(R.id.txtBalance)
    }
}
